Ecal/include/Ecal/Event/EcalWABResult.h

```diff
@@ -0,0 +1,225 @@
+
+  void setRecThetaDiffElectronPhoton(double recThetaDiffElectronPhoton) {
+    recThetaDiffElectronPhoton_ = recThetaDiffElectronPhoton;
+  }
+  double getRecThetaDiffElectronPhoton() const {
+    return recThetaDiffElectronPhoton_;
+  }
+
+  void setRecPhiDiffElectronPhoton(double recPhiDiffElectronPhoton) {
+    recPhiDiffElectronPhoton_ = recPhiDiffElectronPhoton;
+  }
+  double getRecPhiDiffElectronPhoton() const {
+    return recPhiDiffElectronPhoton_;
+  }
+
+  void setTrueRecThetaDiffElectron(double trueRecThetaDiffElectron) {
+    trueRecThetaDiffElectron_ = trueRecThetaDiffElectron;
+  }
+  double getTrueRecThetaDiffElectron() const {
+    return trueRecThetaDiffElectron_;
+  }
+
+  void setTrueRecPhiDiffElectron(double trueRecPhiDiffElectron) {
+    trueRecPhiDiffElectron_ = trueRecPhiDiffElectron;
+  }
+  double getTrueRecPhiDiffElectron() const { return trueRecPhiDiffElectron_; }
+
+  void setTrueRecThetaDiffPhoton(double trueRecThetaDiffPhoton) {
+    trueRecThetaDiffPhoton_ = trueRecThetaDiffPhoton;
+  }
+  double getTrueRecThetaDiffPhoton() const { return trueRecThetaDiffPhoton_; }
+
+  void setTrueRecPhiDiffPhoton(double trueRecPhiDiffPhoton) {
+    trueRecPhiDiffPhoton_ = trueRecPhiDiffPhoton;
+  }
+  double getTrueRecPhiDiffPhoton() const { return trueRecPhiDiffPhoton_; }
+
+  void setTrueElectronShowerEnergy(float trueElectronShowerEnergy) {
+    trueElectronShowerEnergy_ = trueElectronShowerEnergy;
+  }
+  float getTrueElectronShowerEnergy() const {
+    return trueElectronShowerEnergy_;
+  }
+
+  void setTruePhotonShowerEnergy(float truePhotonShowerEnergy) {
+    truePhotonShowerEnergy_ = truePhotonShowerEnergy;
+  }
+  float getTruePhotonShowerEnergy() const { return truePhotonShowerEnergy_; }
+
+  void setRecElectronShowerEnergy(double recElectronShowerEnergy) {
+    recElectronShowerEnergy_ = recElectronShowerEnergy;
+  }
+  double getRecElectronShowerEnergy() const { return recElectronShowerEnergy_; }
+
+  void setRecPhotonShowerEnergy(double recPhotonShowerEnergy) {
+    recPhotonShowerEnergy_ = recPhotonShowerEnergy;
+  }
+  double getRecPhotonShowerEnergy() const { return recPhotonShowerEnergy_; }
+
+  void setProgressNum(int progressNum) { progressNum_ = progressNum; }
+  double getProgressNum() const { return progressNum_; }
+
+ protected:
+  // Actual Kinematic Parameters
+  double trueThetaElectron_;
+  double trueThetaPhoton_;
+  double truePhiElectron_;
+  double truePhiPhoton_;
+  double recThetaElectron_;
+  double recThetaPhoton_;
+  double recPhiElectron_;
+  double recPhiPhoton_;
+  double trueThetaDiffElectronPhoton_;
+  double truePhiDiffElectronPhoton_;
+  double recThetaDiffElectronPhoton_;
+  double recPhiDiffElectronPhoton_;
+  double trueRecThetaDiffElectron_;
+  double trueRecPhiDiffElectron_;
+  double trueRecThetaDiffPhoton_;
+  double trueRecPhiDiffPhoton_;
+  float trueElectronShowerEnergy_;
+  float truePhotonShowerEnergy_;
+  double recElectronShowerEnergy_;
+  double recPhotonShowerEnergy_;
```

Comment:
I think all these should be `float`s, I'm sure we dont need the `double` precision, right? And then we can save some space. This will mean downstream changes that I'm not pointing out rn.

Also the rule I.3. 
https://ldmx-software.github.io/developing/coding-rules.html